### PR TITLE
Amp 45291 ampli js dogfood updates

### PR DESCRIPTION
**Pull request by justin-fiedler**
Updated Ampli JavaScript examples
* Replaced Ampli.getInstance() with ampli.load()
* No longer include `deviceId` as a positional param in identify or getGroup
* Added setGroup
* Node JS SDK now return promise objects
* Added tests in JavaScript examples
* Added Github PR workflow to run Browser & Node JS tests

### CODE REVIEW

File: browser/javascript/react-app/src/ampli.test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  let ampli;
+  let userId = 'test-browser-js-ampli-user-id';
+
+  // Applies only to tests in this describe block
+  beforeEach(() => {
+    ampli = new Ampli();
+    // Set API keys for tests
+    ApiKey.production = 'test-api-key-prod';
+    ApiKey.development = 'test-api-key-dev';
+    // return initializeFoodDatabase();
```
Comment (qingzhuozhen) [inline browser/javascript/react-app/src/ampli.test.js:14]: maybe clean up unused lines?

File: browser/javascript/react-app/src/ampli.test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { Ampli, ApiKey } from './ampli';
+
+describe('Ampli Browser JS SDK tests', () => {
```
Comment (qingzhuozhen) [inline browser/javascript/react-app/src/ampli.test.js:3]: We are adding unit test for generated ampli? Seems some simple ones.

File: browser/typescript/react-app/src/ampli.test.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  let ampli;
+  let userId = 'test-browser-ts-ampli-user-id';
+
+  // Applies only to tests in this describe block
+  beforeEach(() => {
+    ampli = new Ampli();
+    // Set API keys for tests
+    ApiKey.production = 'test-api-key-prod';
+    ApiKey.development = 'test-api-key-dev';
+    // return initializeFoodDatabase();
```
Comment (qingzhuozhen) [inline browser/typescript/react-app/src/ampli.test.js:14]: same clean up

File: node/typescript/src/ampli.test.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  let ampli;
+  let userId = 'test-ampli-user-id';
+
+  // Applies only to tests in this describe block
+  beforeEach(() => {
+    ampli = new Ampli();
+    // Set API keys for tests
+    ApiKey.production = 'test-api-key-prod';
+    ApiKey.development = 'test-api-key-dev';
+    // return initializeFoodDatabase();
```
Comment (qingzhuozhen) [inline node/typescript/src/ampli.test.ts:15]: clean up?

File: yarn.lock [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (qingzhuozhen) [inline yarn.lock:1]: empty file here?

File: browser/javascript/react-app/src/ampli.test.js (hunk 2 above)
Comment (justin-fiedler) [reply browser/javascript/react-app/src/ampli.test.js:3]: Yep, these will allow us to quick run sanity checks after making updates to codegen. We have the snapshot tests in the SG server but they don't verify the generated code works.

File: yarn.lock (hunk 5 above)
Comment (justin-fiedler) [reply yarn.lock:1]: This is a semi-hack for the CI tests `actions/setup-node@v2` runs `yarn install` in the project root and throws a frozen lockfile error if this is removed.

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**